File: recipe-management-domain/src/main/java/fr/cta/recipe/management/domain/mapper/CreateRecipeRequestMapper.java
package fr.cta.recipe.management.domain.mapper;

import fr.cta.recipe.management.domain.entity.Recipe;
import fr.cta.recipe.management.domain.usecase.CreateRecipeUseCase;
import org.mapstruct.Mapper;

@Mapper
public interface CreateRecipeRequestMapper {

    Recipe toRecipe(CreateRecipeUseCase.CreateRecipeInput createRecipeInput);
}

File: recipe-management-domain/src/main/kotlin/fr/cta/recipe/management/domain/mapper/CreateRecipeRequestMapper.kt
package fr.cta.recipe.management.domain.mapper

import org.mapstruct.Mapper
import org.mapstruct.factory.Mappers.getMapper

@Mapper
interface CreateRecipeRequestMapper{
    //    fun toRecipe(createRecipeRequest: CreateRecipeUseCase.CreateRecipeRequest): Recipe
    fun toCarRecipe(carDto: CarDto): Car
}

fun main() {

    val test:CreateRecipeRequestMapper = getMapper(CreateRecipeRequestMapper::class.java)
    val carDto = CarDto()
    carDto.seatCount = 0
    carDto.make = "dsvd"
    carDto.type = "dvsdvssvd"

    print(test.toCarRecipe(carDto))
}